### PR TITLE
Eng 7651 add dr to full ddl

### PR DESCRIPTION
**Pull request by wweiss-voltdb**

### CODE REVIEW

File: src/frontend/org/voltdb/compiler/DDLCompiler.java [hunk 1]
```diff
@@ -968,6 +975,9 @@ private void fillTrackerFromXML()
                 else {
                     m_tracker.removeExportedTable(tableName);
                 }
+                if (drTable != null) {
```
Comment (afrogers) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: All the others remove if it is null -- why not DR, too?

**Comment by afrogers** [comment]
Review complete

File: src/frontend/org/voltdb/compiler/DDLCompiler.java (hunk 1 above)
Comment (wweiss-voltdb) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: Let me double check the behaviour.

File: src/frontend/org/voltdb/compiler/DDLCompiler.java [hunk 2]
```diff
@@ -778,16 +777,23 @@ else if (matchStatus == ClassNameMatchStatus.NO_WILDCARD_MATCH) {
...
-                "Invalid DR statement: table %s is an export table", tableName
-                ));
+            if (tableXML != null) {
+                if (tableXML.attributes.containsKey("export")) {
+                    throw m_compiler.new VoltCompilerException(String.format(
+                        "Invalid DR statement: table %s is an export table", tableName));
+                }
+                else {
+                    if ((statementMatcher.group(2) != null)) {
+                        tableXML.attributes.put("drTable", "DISABLE");
```
Comment (nshi) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: What about just dropping the `drTable` attribute if it's disabled?

**Comment by nshi** [comment]
There is a check in `CatalogDiffEngine.java` that prevents DR table to be exported. I believe that can be removed now with the proper DDL compilation.

**Comment by nshi** [comment]
Except for the first commit, the rest has nothing to do with ENG-7651. Can you edit the commit messages so that they describe what ticket they are fixing? The first commit should contain ENG-7651. Squashing the last three commits would also be appreciated.

File: src/frontend/org/voltdb/compiler/DDLCompiler.java (hunk 2 above)
Comment (wweiss-voltdb) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: It is a style question. The disabled drTable flag translates to the same thing as a missing drTable flag. So if the behaviour is identical, efficiency breaks the tie for me. It is more efficient to do a put than a check followed by a delete.

**Pull request closed without merging**